reversal.go

```diff
@@ -0,0 +1,56 @@
+// Licensed to The Moov Authors under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. The Moov Authors licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ach
+
+import (
+	"time"
+)
+
+// Reversal will transform a File into a Nacha compliant reversal which can be transmitted to undo fund movement.
+func (f *File) Reversal(effectiveEntryDate time.Time) error {
```

Comment:
Would it be worth checking here `if effectiveEntryDate.IsZero()` and returning an error?

Reply:
Perhaps. That'll fail validation as `000000` isn't a valid EffectiveEntryDate value.